lib/mix/test/mix/tasks/compile.erlang_test.exs

```diff
@@ -59,4 +59,30 @@ defmodule Mix.Tasks.Compile.ErlangTest do
       refute File.regular?("_build/dev/lib/sample/ebin/b.beam")
     end
   end
+
+  test "compilation purges the module" do
+    in_fixture "compile_erlang", fn ->
+      File.write! "src/purge_test.erl", """
+      -module(purge_test).
+      -compile(export_all).
+      fn() -> v1.
+      """
+      assert Mix.Tasks.Compile.Erlang.run([]) == :ok
+      # make sure the older version of the module is loaded
+      assert :v1 == :purge_test.fn
+
+      # change the source code
+      File.write! "src/purge_test.erl", """
+      -module(purge_test).
+      -compile(export_all).
+      fn() -> v2.
```

Comment:
`fun` is an Erlang keyword. :bowtie:
I suggest `my_fun` as we usually do `my_` prefixing.